### PR TITLE
feat(FN-692): infrastructure improvements

### PR DESCRIPTION
**Pull request by francescastocco**
## Introduction :pencil2:
Feature environment infrastructure improvements

## Resolution :heavy_check_mark:
- Added what-if to bicep deploy pipeline
- Added a pipeline for linting bicep code when pull request created against `feat/stbr-infra`
- Disabled the function-apps deployment in the feature_deploy pipeline

## Miscellaneous :heavy_plus_sign:
List any additional fixes or improvements.

### CODE REVIEW

File: .github/workflows/bicep_lint.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+on:
+  pull_request:
+    branches:
+      - feat/stbr-infra
+    paths:
+      - '.github/workflows/bicep_az_deployment_group.yml'
+      - '.github/workflows/bicep_deploy.yml'
+      - '.github/workflows/bicep_workflow_feature.yml'
+      - 'infrastructure/resource_group_level/**'
```
Comment (ChristianMcc) [inline .github/workflows/bicep_lint.yml:11]: should this just be infrastructure? Right now I assume everything is resource_group_level but would assume if we ever have any bicep that's at a subscription level we would need it here

File: .github/workflows/bicep_deploy.yml [hunk 2]
```diff
@@ -127,9 +127,18 @@ jobs:
...
+    uses: ./.github/workflows/bicep_az_deployment_group.yml
+    with:
+      environment: ${{ inputs.environment }}
+      resourceGroupName: ${{ inputs.resourceGroupName }}
+      action: what-if
+    secrets: inherit
 
   deploy:
-    needs: validate
+    needs: what_if
```
Comment (ChristianMcc) [inline .github/workflows/bicep_deploy.yml:141]: was there a way for this to be manual only for certain environments and not for others?

File: .github/workflows/bicep_lint.yml (hunk 1 above)
Comment (francescastocco) [reply .github/workflows/bicep_lint.yml:11]: Hm yes, good point

File: .github/workflows/bicep_deploy.yml (hunk 2 above)
Comment (francescastocco) [reply .github/workflows/bicep_deploy.yml:141]: I've left it as automatic for all environments for now as it looked quite complex to add the manual step so will come back and look at it if I get more time

Comment (ChristianMcc) [reply .github/workflows/bicep_deploy.yml:141]: Sounds good, will resolve :)

**Review by ChristianMcc — APPROVED** [review]

**Pull request merged**